var/spack/repos/builtin/packages/py-jupyterlab/package.py

```diff
@@ -50,12 +50,9 @@ class PyJupyterlab(PythonPackage):
 
     with when("@:3"):
         depends_on("py-setuptools", when="@:3", type=("build", "run"))
-        # TODO: replace this after concretizer learns how to concretize separate build deps
-        depends_on("py-jupyter-packaging11", when="@3.0.15:3", type="build")
-        depends_on("py-jupyter-packaging7", when="@3.0.0:3.0.14", type="build")
-        # depends_on('py-jupyter-packaging@0.9:0', when='@3.0.15:', type='build')
-        # depends_on('py-jupyter-packaging@0.7.3:0.7', when='@3.0.0:3.0.14',
-        #            type=('build', 'run'))
+        depends_on("py-jupyter-packaging@0.9:1", when="@3.4.8", type="build")
+        depends_on("py-jupyter-packaging@0.9:0", when="@3.0.15:3.4.2", type="build")
+        depends_on("py-jupyter-packaging@0.7.3:0.7", when="@3.0.0:3.0.14", type=("build", "run"))
```

Comment:
Why `build` + `run` in this line?

Reply:
Because it's (erroneously) listed under `install_requires` instead of `setup_requires`: https://github.com/jupyterlab/jupyterlab/blob/v3.0.14/setup.py#L149. This was fixed in https://github.com/jupyterlab/jupyterlab/pull/10217 and backported to 3.0.15.

If this is problematic for concretization we _could_ make it build-only or even deprecate and remove 3.0.14.

Reply:
If we can deprecate and remove that version, that would be good. This one is not problematic per-se, but it would be if more than one needed version is of type "run".